src/components/NcRichContenteditable/NcRichContenteditable.vue

```diff
@@ -212,28 +213,38 @@ export default {
 </docs>
 
 <template>
-	<div ref="contenteditable"
-		v-tooltip="tooltipString"
-		:class="{
-			'rich-contenteditable__input--empty': isEmptyValue,
-			'rich-contenteditable__input--multiline': multiline,
-			'rich-contenteditable__input--overflow': isOverMaxlength,
-			'rich-contenteditable__input--disabled': disabled,
-		}"
-		:contenteditable="canEdit"
-		:aria-placeholder="placeholder"
-		aria-multiline="true"
-		class="rich-contenteditable__input"
-		role="textbox"
-		v-on="listeners"
-		@input="onInput"
-		@compositionstart="isComposing = true"
-		@compositionend="isComposing = false"
-		@keydown.delete="onDelete"
-		@keydown.enter.exact="onEnter"
-		@keydown.ctrl.enter.exact.stop.prevent="onCtrlEnter"
-		@paste="onPaste"
-		@keyup.stop.prevent.capture="onKeyUp" />
+	<div class="rich-contenteditable">
+		<div :id="id"
+			ref="contenteditable"
+			v-tooltip="tooltipString"
+			:class="{
+				'rich-contenteditable__input--empty': isEmptyValue,
+				'rich-contenteditable__input--multiline': multiline,
+				'rich-contenteditable__input--has-label': label,
+				'rich-contenteditable__input--overflow': isOverMaxlength,
+				'rich-contenteditable__input--disabled': disabled,
+			}"
+			:contenteditable="canEdit"
+			:aria-placeholder="placeholder"
+			aria-multiline="true"
+			class="rich-contenteditable__input"
+			role="textbox"
+			v-bind="$attrs"
+			v-on="listeners"
+			@input="onInput"
+			@compositionstart="isComposing = true"
+			@compositionend="isComposing = false"
+			@keydown.delete="onDelete"
+			@keydown.enter.exact="onEnter"
+			@keydown.ctrl.enter.exact.stop.prevent="onCtrlEnter"
+			@paste="onPaste"
+			@keyup.stop.prevent.capture="onKeyUp" />
+		<label v-if="label"
+			class="rich-contenteditable__label"
+			:for="id">
+			{{ label }}
+		</label>
```

Comment:
I haven't checked all screen readers' support, but according to the standard, it is not valid to make a label for a div. We should use `aria-labelledby` here.

See: https://html.spec.whatwg.org/multipage/forms.html#category-label

Reply:
Yes MDN also recommends using labelledby. Changed that.